### ISSUE TITLE
SignalR Extensions: enable network requests to be routed through any application gateway.

### ISSUE THREAD

**Issue by cmendible** [Azure/azure-sdk-for-net#26744]
### Library name

Microsoft.Azure.WebJobs.Extensions.SignalRService

### Please describe the feature.

SignalR Service extensions should have a way to set the the address to use for establishing a connection to the SignalR Service, allowing network requests to be routed through any application gateways or other paths needed for the host environment.

EventHub extensions enable this scenario via the CustomEndpointProperty:

https://github.com/Azure/azure-sdk-for-net/blob/07f98ef8d32cab0ee4aeadf5b32c1014451f2cb5/sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Config/EventHubOptions.cs#L78

**Comment by SaurabhSharma-MSFT** [comment]
@cmendible Redirecting this to service team to provide you alternatives if any.

**Comment by chenkennt** [comment]
@Y-Sindo please help take a look

**Comment by Y-Sindo** [comment]
Azure SignalR currently supports client endpoint in the connection strings as follows: 
```
<connection-string-of-Azure-SignalR>;ClientEndpoint=http://<public-ip-of-AG1>
```

**Comment by cmendible** [comment]
> ClientEndpoint

That's great! Searching for `Azure SignalR + ClientEndpoint` I just found this post: https://techcommunity.microsoft.com/t5/apps-on-azure-blog/protect-your-signalr-service-based-real-time-application-with/ba-p/2589650 with a complete sample for this scenario.

Is there a reference in **docs** to `ClientEndpoint` in the connection string?

**Comment by Y-Sindo** [comment]
The new feature is released in the 1.8.0 version. [Go here for more details.](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/signalr/Microsoft.Azure.WebJobs.Extensions.SignalRService/samples/Sample03_IntegrationWithAppGateway.md)
